src/components/safe-messages/MsgModal/index.tsx

```diff
@@ -147,96 +189,143 @@ const MsgModal = ({
+            </Typography>
+
+            <Typography variant="body2" fontWeight={700} mt={2}>
+              SafeMessage hash:
+            </Typography>
+            <Typography variant="body2">
+              <EthHashInfo address={safeMessageHash} showAvatar={false} shortAddress={false} showCopyButton />
+            </Typography>
 
+            {hasSigned && (
+              <SuccessMessage>
+                <Grid container direction="row">
+                  <Grid item xs>
+                    Your connected wallet has already signed this message.
+                  </Grid>
+                  <Grid item xs={4}>
+                    <Button variant="contained" size="small" onClick={handleSwitchWallet} fullWidth>
+                      Switch wallet
+                    </Button>
+                  </Grid>
+                </Grid>
+              </SuccessMessage>
+            )}
+
+            <InfoBox
+              message={
+                requestId
+                  ? 'Please keep this modal open until all signers will confirm the message. Closing this modal will abort the signing request.'
+                  : 'The first signer will submit the siganture to the Safe App when the message is fully signed'
+              }
+            >
+              <MsgSigners
+                msg={ongoingMessage || createSkeletonMessage(safe.threshold)}
+                showOnlyConfirmations
+                showMissingSignatures
+                backgroundColor={palette.info.background}
+              />
+            </InfoBox>
+
+            {/* Warning message and switch button */}
+            <WrongChainWarning />
+
+            {!wallet || !onboard ? (
+              <ErrorMessage>No wallet is connected.</ErrorMessage>
+            ) : !isOwner ? (
+              <ErrorMessage>
+                You are currently not an owner of this Safe Account and won&apos;t be able to confirm this message.
+              </ErrorMessage>
+            ) : submitError ? (
+              <ErrorMessage error={submitError}>Error confirming the message. Please try again.</ErrorMessage>
+            ) : null}
+          </DialogContent>
+
+          <DialogActions>
+            <Button onClick={handleClose}>Cancel</Button>
+            <Button variant="contained" color="inherit" onClick={onSign} disabled={isDisabled}>
+              Sign
+            </Button>
+          </DialogActions>
+        </div>
+      </ModalDialog>
+      {/* Confirmation Dialog */}
+      <Dialog
+        open={showCloseTooltip}
+        onClose={handleClose}
+        aria-labelledby="alert-dialog-title"
+        aria-describedby="alert-dialog-description"
+      >
+        <DialogTitle id="alert-dialog-title">Cancel sign message request</DialogTitle>
+        <DialogContent>
+          <DialogContentText id="alert-dialog-description">
+            <Typography variant="body2">If you close the modal now, the signature request will be aborted.</Typography>
```

Comment:
We may want to adjust this text/that on line 274 to match. What do you think?

```suggestion
            <Typography variant="body2">If you close this modal, the signing request will be aborted.</Typography>
```